policy/modules/services/node_exporter.te

```diff
@@ -0,0 +1,62 @@
+policy_module(node_exporter)
+
+type node_exporter_t;
+type node_exporter_exec_t;
+init_daemon_domain(node_exporter_t, node_exporter_exec_t)
+
+type node_exporter_runtime_t;
+files_runtime_file(node_exporter_runtime_t)
+
+type node_exporter_var_lib_t;
+files_type(node_exporter_var_lib_t)
+
+type node_exporter_var_log_t;
+logging_log_file(node_exporter_var_log_t)
+
+allow node_exporter_t self:fifo_file rw_fifo_file_perms;
+allow node_exporter_t self:process { getsched signal };
+allow node_exporter_t self:netlink_route_socket r_netlink_socket_perms;
+allow node_exporter_t self:tcp_socket create_stream_socket_perms;
+allow node_exporter_t self:udp_socket create_socket_perms;
+
+manage_files_pattern(node_exporter_t, node_exporter_runtime_t, node_exporter_runtime_t)
+files_runtime_filetrans(node_exporter_t, node_exporter_runtime_t, file)
+
+manage_dirs_pattern(node_exporter_t, node_exporter_var_lib_t, node_exporter_var_lib_t)
+manage_files_pattern(node_exporter_t, node_exporter_var_lib_t, node_exporter_var_lib_t)
+files_var_lib_filetrans(node_exporter_t, node_exporter_var_lib_t, { dir file })
+
+append_files_pattern(node_exporter_t, node_exporter_var_log_t, node_exporter_var_log_t)
+create_files_pattern(node_exporter_t, node_exporter_var_log_t, node_exporter_var_log_t)
+setattr_files_pattern(node_exporter_t, node_exporter_var_log_t, node_exporter_var_log_t)
+logging_log_filetrans(node_exporter_t, node_exporter_var_log_t, { dir file })
+
+corenet_tcp_bind_all_ports(node_exporter_t)
```

Comment:
This seems far too broad. Isn't there some reasonable subset of this?

Reply:
The default port for node_exporter is `9100/tcp` which is currently assigned to `hplip_t`. We can just replace this with `corenet_tcp_bind_hplip_port()` but it feels wrong. Otherwise it accomplishes the intended goal without giving too much access.

Reply:
The `node_exporter` folks indeed decided to use a reserved port (and `hplip` also has more ports than would be usual). I picked the compromise of having it be able to listen on any port and then rely on the firewall to control access.

Reply:
I think it's OK to use hplip here, as long as there is a comment that says the node exporter also uses 9100. Hplip came first.